src/app/wallets/add-invoice-for-wallet.ts

```diff
@@ -1,88 +1,61 @@
-import { checkedToSats, toSats } from "@domain/bitcoin"
-import { invoiceExpirationForCurrency } from "@domain/bitcoin/lightning"
-import { defaultTimeToExpiryInSeconds } from "@domain/bitcoin/lightning/invoice-expiration"
-import { DealerPriceServiceError } from "@domain/dealer-price"
-import { checkedtoCents } from "@domain/fiat"
 import { RateLimitConfig } from "@domain/rate-limit"
 import { RateLimiterExceededError } from "@domain/rate-limit/errors"
-import { WalletInvoiceFactory } from "@domain/wallet-invoices/wallet-invoice-factory"
 import { checkedToWalletId } from "@domain/wallets"
-import { WalletCurrency } from "@domain/shared"
-import { DealerPriceService } from "@services/dealer-price"
+import { NewDealerPriceService } from "@services/dealer-price"
 import { LndService } from "@services/lnd"
 import { WalletInvoicesRepository, WalletsRepository } from "@services/mongoose"
 import { consumeLimiter } from "@services/rate-limit"
+import { WalletInvoiceBuilder } from "@domain/wallet-invoices/wallet-invoice-builder"
 
 export const addInvoiceForSelf = async ({
```

Comment:
@bodymindarts How does this look for deduplicating the code in the different use cases?